cmd/fyne_demo/screens/widget.go

```diff
@@ -116,12 +116,14 @@ func makeScrollTab() fyne.CanvasObject {
 		}))
 	}
 
-	horiz := widget.NewScrollContainer(list)
-	vert := widget.NewScrollContainer(list2)
-
-	scrolls := fyne.NewContainerWithLayout(layout.NewBorderLayout(horiz, nil, nil, nil),
-		horiz, vert)
-	return fyne.NewContainerWithLayout(layout.NewAdaptiveGridLayout(2), scrolls, makeScrollBothTab())
+	horiz := widget.NewHorizontalScrollContainer(list)
+	vert := widget.NewVerticalScrollContainer(list2)
+	vert.SetMinSize(fyne.NewSize(32, 128))
+
+	return fyne.NewContainerWithLayout(layout.NewAdaptiveGridLayout(2), widget.NewVBox(
```

Comment:
Oh I see why you are having problems with VBox now - why don't you use the container/layout that was already here that did not suffer the problem you describe?

Reply:
That's fine for this case, but adding a scroller to a borderlayout to force it to take up all the space is not a solution that can be applied to all use cases. We need something better.

In general, layouts measure the min size of their objects, but if that layout is invoked to layout objects in a bigger size, shouldn't it expand its objects to take up all the space? Granted there are cases you wouldn't want this, eg toolbars should have a max height.

Reply:
Any widget that is expanded can ask it's content to expand. HBox and VBox are special in that they pack items to the left, or the top. That is how they are designed and so by moving to use VBox you require the workaround above. I don't think there is a good reason to keep it this way.

If you feel layouts need improvements please open a new issue so we can discuss separately.